### ISSUE TITLE
Compose a post with a question mark prompts shortcut modal on publish

### ISSUE THREAD

**Issue by djwfyi** [elk-zone/elk#1910]
This is happening to me frequently.

## Description

When I compose a post that includes a question mark, i'm having to select the **Publish** button twice.
The first time I select it, the keyboard shortcut modal displays.
I have to dismiss the modal, then select the **Publish** button a second time to get the post to publish.

## Steps to reproduce

1. In the Compose window on a desktop, type a post with a question mark

   I'm not sure yet if the question mark has to be the last character or anywhere in the post.

1. Select the **Publish** button

   Keyboard shortcut modal displays (since the `?` is a shortcut to display it).

1. Dismiss modal
1. Select the **Publish** button again

   Post publishes.

## What should happen?

Post should publish the first time I select the **Publish** button without displaying the shortcut modal.

## System Info

- Elk version v0.7.5 · [c76e8e8](https://github.com/elk-zone/elk/commit/c76e8e8f5eb27d3f24e0359a532809adaeb463ac)
- Host: elk.vmst.io
- OS: Pop!_OS 22.04
- Browser: Firefox 111.0

**Comment by johnsonm** [comment]
Thank you for filing this. I hadn't figured out what was triggering this sporadic problem until I found this bug report!

**Comment by djwfyi** [comment]
@johnsonm This seems to be happening if you type a `?` any time during post composition, even if it doesn't make it to the final post. Elk should ignore the keyboard shortcuts while the post compose box is active. It seems like it is not at the moment.

**Comment by johnsonm** [comment]
This is definitely still happening on main, and I can confirm what @djwfyi reports that it happens even if the `?` was ever typed, even if it was subsequently erased from the box.

**Comment by johnsonm** [comment]
Is this by any chance specific to Firefox?

**Comment by johnsonm** [comment]
Mine too.

I asked because I figured this is so annoying that the developers must not be seeing it because they are using Chrome rather than Firefox, but I just reproduced this problem in Chrome, so it is not specific to Firefox. So a wrong assumption and wrong guess on my part.

**Comment by patak-dev** [comment]
I can't reproduce this in Chrome or Firefox on MacOS. Would you try using https://main.elk.zone so you are testing the latest version? If not, maybe send a message to https://chat.elk.zone to see if others can reproduce this 🤔

**Comment by djwfyi** [comment]
@patak-dev 
I don't see it on main.elk.zone running Canary, either, but still happening on elk.vmst.io running Alpha / v0.8.0.

**Comment by patak-dev** [comment]
Interesting, I don't see any related commit since v0.8.0. We can check elk.vmst.io after the next release.

**Comment by djwfyi** [comment]
Ah! I did reproduce it on main.elk.zone.
It's not the compose window when you click on the Compose link in the sidebar.

It's when you reply to a post that it happens.
So click to comment on a post and then type something with a question. I got it to happen on main.elk.zone.
@patak-dev

**Comment by patak-dev** [comment]
I still can't reproduce it 🤔 

Just in case, I'm testing this one:
Built 2 days ago · canary · [d34a5e6](https://github.com/elk-zone/elk/commit/d34a5e6e9638b1d39bb77b9a8f7732908363bcf4)

**Comment by djwfyi** [comment]
Just for reference: MacOS Ventura 13.3.1, Firefox 112.0.1

**Comment by johnsonm** [comment]
I run only main.elk.zone, updating the app when prompted, and have never failed to reproduce it in about the past month. I'm using Linux with GNOME and Wayland.

I compose either from the top of the home view or in a reply to an existing post, and it shows up consistently in both contexts.

**Comment by johnsonm** [comment]
I can reproduce what I think is essentially the same by typing `?` and then clicking the visibility icon, without posting. 

I've reproduced this on Firefox 110 and 112 as well as Chrome 112.

**Comment by johnsonm** [comment]
I note that both @djwfyi and I are running Linux. I'm running Fedora 37 with Wayland.

**Comment by djwfyi** [comment]
I've reproduced this on Pop, MacOS Ventura, and both Windows 10 and 11. I've encountered it in Firefox and Chrome.

**Comment by pjohanneson** [comment]
It's happening to me too. I'm using Firefox 112.0 on Ubuntu 22.04, and it happened on main.elk.zone canary · [1ceb3e2](https://github.com/elk-zone/elk/commit/1ceb3e2857e05ac1d03d1038e3b70bc1350f72ca)

**Comment by alexzhang1030** [comment]
Do you have a reproduction video? It can help us quickly track this issue.

**Comment by johnsonm** [comment]
I saw this as recently as (I think) yesterday, but I went to record a screen cast of it just now and it did not reproduce. (main.elk.zone)

**Comment by alexzhang1030** [comment]
> I saw this as recently as (I think) yesterday, but I went to record a screen cast of it just now and it did not reproduce. (main.elk.zone)

Let's keep tracking, I cannot reproduce it on my device.

**Comment by djwfyi** [comment]
Here's a screen recording I just took.

https://user-images.githubusercontent.com/53910321/235565045-f27100a9-fb72-476b-b96a-f968ff99db41.mp4

Instance: elk.vmst.io, running v0.8.1
OS: Pop!_OS 22.04, GNOME 42.5, X11
Browser: Firefox 112.0.1

Steps:
1. From the Home timeline, click on the reply button to reply to a thread
2. In the reply compose box, enter any text that includes a `?`. (You can go on to delete the `?` if you want.)
3. Select the **Publish** button
4. See that the keyboard shortcut window displays.

NOTE: This does NOT happen reliably in the standard compose window. But does happen reliably from the reply compose window.

**Comment by djwfyi** [comment]
I tried this on Chromium as well. First time I replied to a post with a question mark, it did not happen.
The second time I replied to a post with a question mark, it did happen.

That's more than a little frustrating that it didn't happen the first time. But it does show that it's not unique to Firefox.

**Comment by johnsonm** [comment]
This happened to me again in normal use from the standard compose window, so I tried twice to record a screencast of it happening, and it didn't happen either time. Then as @djwfyi pointed out, I tried a reply window. Instead of the "Reply" button, I pressed the globe visibility button, which reproduced the problem.

[Screencast from 2023-05-05 20-43-55.webm](https://user-images.githubusercontent.com/79751/236589212-47327350-6c80-4a55-b30e-9e75e2b1caeb.webm)

**Comment by joelanman** [comment]
Happens for me intermittently on Safari, MacOS

**Comment by johnsonm** [comment]
While this problem became less prevalent for a while, it has now started happening again regularly.

**Comment by userquin** [comment]
Trying to fix this

**Comment by userquin** [comment]
> While this problem became less prevalent for a while, it has now started happening again regularly.

Can you check it with the PR preview?

**Comment by johnsonm** [comment]
Am I supposed to be automatically logged in as `@elkdev@universeodon.com` just by running the preview?

I logged out of elkdev and into my own account and I confirm that so far I have not managed to reproduce this problem with this fix. :heart:

**Comment by johnsonm** [comment]
For those others here who would like to test on their browser/OS combo, just go to https://deploy-preview-2209--elk-zone.netlify.app/

**Comment by userquin** [comment]
> Am I supposed to be automatically logged in as @elkdev@universeodon.com just by running the preview?

yeah, we mock the users and so we don't need to sign-in to test it, you can just sign-in to your accounts, there is no need to sign-out from ElkDev user.

**Comment by pjohanneson** [comment]
Sure looks fixed to me on main.elk.zone (https://github.com/elk-zone/elk/commit/ae35f9d11d67a1b8f280391486b4e4a905a67d53), thanks!
